tests/foreman/ui_airgun/test_contentcredential.py

```diff
@@ -24,712 +24,864 @@
+        content=gpg_content,
+        organization=module_org,
+    ).create()
+    with session:
+        session.product.create(
+            {'name': prod_name, 'gpg_key': gpg_key.name})
+        values = session.contentcredential.read(gpg_key.name)
+        assert len(values['products']['table']) == 1
+        assert values['products']['table'][0]['Name'] == prod_name
+        assert values['products']['table'][0]['Used as'] == 'GPG Key'
+
+
+@tier2
+def test_positive_add_product_with_repo(session, module_org, gpg_content):
+    """Create gpg key with valid name and valid gpg key then associate it
+    with custom product that has one repository
+
+    :id: 7514b33a-da75-43bd-a84b-5a805c84511d
+
+    :expectedresults: gpg key is associated with product as well as with
+        the repository
+
+    :CaseLevel: Integration
+    """
+    name = gen_string('alpha')
+    gpg_key = entities.GPGKey(
+        content=gpg_content,
+        name=name,
+        organization=module_org,
+    ).create()
+    # Creates new product and associate GPGKey with it
+    product = entities.Product(
+        gpg_key=gpg_key,
+        organization=module_org,
+    ).create()
+    # Creates new repository without GPGKey
+    repo = entities.Repository(
+        url=FAKE_1_YUM_REPO,
+        product=product,
+    ).create()
+    with session:
+        values = session.contentcredential.read(name)
+        assert len(values['products']['table']) == 1
+        assert values['products']['table'][0]['Name'] == product.name
+        assert len(values['repositories']['table']) == 1
+        assert values['repositories']['table'][0]['Name'] == repo.name
+        assert (
+            values['repositories']['table'][0]['Product'] ==
+            product.name
+        )
+        assert values['repositories']['table'][0]['Type'] == 'yum'
+        assert (
+            values['repositories']['table'][0]['Used as'] ==
+            'GPG Key'
+        )
+
+
+@tier2
+def test_positive_add_product_with_repos(session, module_org, gpg_content):
+    """Create gpg key with valid name and valid gpg key then associate it
+    with custom product that has more than one repository
+
+    :id: 0edffad7-0ab4-4bef-b16b-f6c8de55b0dc
+
+    :expectedresults: gpg key is properly associated with repositories
+
+    :CaseLevel: Integration
+    """
+    name = gen_string('alpha')
+    gpg_key = entities.GPGKey(
+        content=gpg_content,
+        name=name,
+        organization=module_org,
+    ).create()
+    # Creates new product and associate GPGKey with it
+    product = entities.Product(
+        gpg_key=gpg_key,
+        organization=module_org,
+    ).create()
+    # Creates new repository_1 without GPGKey
+    repo1 = entities.Repository(
+        product=product,
+        url=FAKE_1_YUM_REPO,
+    ).create()
+    # Creates new repository_2 without GPGKey
+    repo2 = entities.Repository(
+        product=product,
+        url=FAKE_2_YUM_REPO,
+    ).create()
+    with session:
```

Comment:
Same as with one repo, think there is a divergence between the test requirement and the implementation.